### ISSUE TITLE
The artifact v0.4.2 for Scala.js is broken

### ISSUE THREAD

**Issue by sjrd** [lihaoyi/scala.rx#171]
When trying to use scala.rx 0.4.2 with Scala.js 1.1.0, I get the following error at link time:
```
sbt:hands-on-scala-js> demos/fastOptJS
[warn] There may be incompatibilities among your library dependencies; run 'evicted' to see detailed eviction warnings.
[info] Compiling 20 Scala sources to C:\Users\sjrdo\Documents\Projets\hands-on-scala-js\examples\demos\target\scala-2.12\classes ...
[info] Fast optimizing C:\Users\sjrdo\Documents\Projets\hands-on-scala-js\examples\demos\target\scala-2.12\example-fastopt.js
[error] stack trace is suppressed; run last demos / Compile / fastOptJS for the full output
[error] (demos / Compile / fastOptJS) java.io.IOException: Failed to deserialize C:\Users\sjrdo\AppData\Local\Coursier\cache\v1\https\repo1.maven.org\maven2\com\lihaoyi\scalarx_sjs1_2.12\0.4.2\scalarx_sjs1_2.12-0.4.2.jar:/rx/opmacros/Factories$$typecreator9$1.sjsir
[error] Total time: 7 s, completed 27 mai 2020 20:56:15
sbt:hands-on-scala-js> last
[debug] > Exec(demos/fastOptJS, Some(8867bb32-dd90-4517-816f-16fef17efd25), Some(CommandSource(console0)))
[debug] Evaluating tasks: checkBuildSources
[debug] Running task... Cancel: bloop.integrations.sbt.Offloader$$anon$1@737e4a7b, check cycles: false, forcegc: true
[debug] Global IR cache stats: reused: 0 -- invalidated: 0 -- trees read: 0
[debug] Evaluating tasks: demos / Compile / fastOptJS
[debug] Running task... Cancel: bloop.integrations.sbt.Offloader$$anon$1@309e874, check cycles: false, forcegc: true
[warn] There may be incompatibilities among your library dependencies; run 'evicted' to see detailed eviction warnings.
[info] Compiling 20 Scala sources to C:\Users\sjrdo\Documents\Projets\hands-on-scala-js\examples\demos\target\scala-2.12\classes ...
[info] Fast optimizing C:\Users\sjrdo\Documents\Projets\hands-on-scala-js\examples\demos\target\scala-2.12\example-fastopt.js
[debug] Global IR cache stats: reused: 20 -- invalidated: 48 -- trees read: 0
[error] java.io.IOException: Failed to deserialize C:\Users\sjrdo\AppData\Local\Coursier\cache\v1\https\repo1.maven.org\maven2\com\lihaoyi\scalarx_sjs1_2.12\0.4.2\scalarx_sjs1_2.12-0.4.2.jar:/rx/opmacros/Factories$$typecreator9$1.sjsir
[error]         at org.scalajs.linker.interface.unstable.IRFileImpl$$anonfun$withPathExceptionContext$1.applyOrElse(IRFileImpl.scala:65)
[error]         at org.scalajs.linker.interface.unstable.IRFileImpl$$anonfun$withPathExceptionContext$1.applyOrElse(IRFileImpl.scala:58)
[error]         at scala.runtime.AbstractPartialFunction.apply(AbstractPartialFunction.scala:38)
[error]         at scala.util.Failure.recover(Try.scala:234)
[error]         at scala.concurrent.Future.$anonfun$recover$1(Future.scala:395)
[error]         at scala.concurrent.impl.Promise.liftedTree1$1(Promise.scala:33)
[error]         at scala.concurrent.impl.Promise.$anonfun$transform$1(Promise.scala:33)
[error]         at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:64)
[error]         at java.util.concurrent.ForkJoinTask$RunnableExecuteAction.exec(ForkJoinTask.java:1402)
[error]         at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289)
[error]         at java.util.concurrent.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1056)
[error]         at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1692)
[error]         at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:157)
[error] Caused by: java.lang.IllegalArgumentException: malformed version: 0.6.17
[error]         at org.scalajs.ir.VersionChecks$.$anonfun$mustMatch$1(ScalaJSVersions.scala:99)
[error]         at scala.Option.getOrElse(Option.scala:189)
[error]         at org.scalajs.ir.VersionChecks$.mustMatch(ScalaJSVersions.scala:99)
[error]         at org.scalajs.ir.VersionChecks$.org$scalajs$ir$VersionChecks$$parseBinary(ScalaJSVersions.scala:88)
[error]         at org.scalajs.ir.VersionChecks.checkSupported(ScalaJSVersions.scala:63)
[error]         at org.scalajs.ir.Serializers$Deserializer.readHeader(Serializers.scala:994)
[error]         at org.scalajs.ir.Serializers$Deserializer.deserializeEntryPointsInfo(Serializers.scala:953)
[error]         at org.scalajs.ir.Serializers$.$anonfun$deserializeEntryPointsInfo$1(Serializers.scala:52)
[error]         at org.scalajs.ir.Serializers$.withBigEndian(Serializers.scala:67)
[error]         at org.scalajs.ir.Serializers$.deserializeEntryPointsInfo(Serializers.scala:52)
[error]         at org.scalajs.linker.standard.MemIRFileImpl.$anonfun$entryPointsInfo$1(MemIRFile.scala:31)
[error]         at org.scalajs.linker.standard.MemIRFileImpl.$anonfun$withBuffer$1(MemIRFile.scala:39)
[error]         at scala.concurrent.Future$.$anonfun$apply$1(Future.scala:659)
[error]         at scala.util.Success.$anonfun$map$1(Try.scala:255)
[error]         at scala.util.Success.map(Try.scala:213)
[error]         at scala.concurrent.Future.$anonfun$map$1(Future.scala:292)
[error]         at scala.concurrent.impl.Promise.liftedTree1$1(Promise.scala:33)
[error]         at scala.concurrent.impl.Promise.$anonfun$transform$1(Promise.scala:33)
[error]         at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:64)
[error]         at java.util.concurrent.ForkJoinTask$RunnableExecuteAction.exec(ForkJoinTask.java:1402)
[error]         at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289)
[error]         at java.util.concurrent.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1056)
[error]         at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1692)
[error]         at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:157)
[error] (demos / Compile / fastOptJS) java.io.IOException: Failed to deserialize C:\Users\sjrdo\AppData\Local\Coursier\cache\v1\https\repo1.maven.org\maven2\com\lihaoyi\scalarx_sjs1_2.12\0.4.2\scalarx_sjs1_2.12-0.4.2.jar:/rx/opmacros/Factories$$typecreator9$1.sjsir
[error] Total time: 7 s, completed 27 mai 2020 20:56:15
[debug] > Exec(shell, None, None)
[debug] Evaluating tasks: checkBuildSources
[debug] Running task... Cancel: bloop.integrations.sbt.Offloader$$anon$1@3ce17e4, check cycles: false, forcegc: true
[debug] Global IR cache stats: reused: 0 -- invalidated: 0 -- trees read: 0
```
Note the error
```
Caused by: java.lang.IllegalArgumentException: malformed version: 0.6.17
```
And sure enough, the file `rx/opsmacros/Factories$$typecreator9$1.sjsir` in the .jar published for Scala.js 1.x has the IR version 0.6.17.

Something must have gone wrong during the publishing. Perhaps the repo was not `clean` enough, and some compilation products of 0.6.x were still on the file system, and made their way to the published jar.

Would you mind publishing a new version from a clean build.

I will need this to update the [Hands-on Scala.js book](https://www.lihaoyi.com/hands-on-scala-js/) for Scala.js 1.x. It is a blocker for me at the moment.

**Comment by sjrd** [comment]
This happens only in the artifact for Scala 2.12. I was able to work around the issue by upgrading to 2.13. The artifact for 2.13 does not have the offending .sjsir file.

**Comment by fdietze** [comment]
@Voltir has publishing permissions. :pray:

**Comment by Voltir** [comment]
@sjrd @fdietze Just published a new/clean build under 0.4.3 - I hope that works. Also, thanks for the amazing work that is scala.js, @sjrd :)

**Comment by sjrd** [comment]
Thank you @Voltir !
